src/mmw/js/src/modeling/models.js

```diff
@@ -370,6 +370,28 @@ var ScenarioModel = Backbone.Model.extend({
         }
     },
 
+    setResults: function() {
+        var rawServerResults = this.get('taskModel').get('result');
+        if (rawServerResults === "" || rawServerResults === null) {
+            this.get('results').setNullResults();
+        } else {
+            var serverResults = JSON.parse(rawServerResults);
+            this.get('results').forEach(function(resultModel) {
+                var resultName = resultModel.get('name');
+                if (serverResults[resultName]) {
+                    resultModel.set({
+                        'result': serverResults[resultName],
+                        'inputmod_hash': serverResults.inputmod_hash
+                    });
+                } else {
+                    console.log('Response is missing ' + resultName + '.');
+                }
+            });
+
+            this.set('census', serverResults.census);
+        }
+    },
+
     // Poll the taskModel for results and reset the results collection when done.
     // If not successful, the results collection is reset to be empty.
     getResults: function() {
```

Comment:
This might be a big ask, since this function is so pervasive and may require touching many files, but —

Since the value returned by `getResults` is not a ResultsCollection but a promise, consider renaming to `fetchResults` instead.